components/script/dom/offscreencanvas.rs

```diff
@@ -142,6 +147,85 @@ impl OffscreenCanvas {
     }
 }
 
+impl Transferable for OffscreenCanvas {
+    type Index = OffscreenCanvasIndex;
+    type Data = TransferableOffscreenCanvas;
+
+    fn can_transfer(&self) -> bool {
+        // TODO (#37882): Allow to transfer with a placeholder canvas element.
+        self.placeholder.is_none()
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#the-offscreencanvas-interface:offscreencanvas-16>
+    fn transfer(&self) -> Result<(OffscreenCanvasId, TransferableOffscreenCanvas), ()> {
+        // Step 1. If value's context mode is not equal to none, then throw an
+        // "InvalidStateError" DOMException.
+        if !self.context.borrow().is_none() {
+            return Err(());
+        }
+
+        // Step 2. Set value's context mode to detached.
+        *self.context.borrow_mut() = Some(OffscreenRenderingContext::Detached);
+
+        // Step 3. Let width and height be the dimensions of value's bitmap.
+        let width = self.width.get();
+        let height = self.height.get();
```

Comment:
You can use https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.replace here, and add a note that this also implements step 5 below(see other comment).

Reply:
ok, thanks